pkg/console/network.go

```diff
@@ -107,6 +126,19 @@ func applyNetworks(network config.Network, hostname string) ([]byte, error) {
 	if err := upAllLinks(); err != nil {
 		logrus.Errorf("failed to bring all link up: %s", err.Error())
 	}
+
+	if hostname == "" && previousHostname != "" {
+		currentHostname, err := getHostname()
+		if err == nil && currentHostname == "localhost" {
+			// Restore the system hostname to what it was before, provided
+			// we haven't got a new hostname from the DHCP server.
+			err = setHostname(previousHostname)
+		}
+		if err != nil {
```

Comment:
if there was an error in getting `currentHostname` the message returned would get confusing

not sure if we should handle the `setHostname` call error in the inner code block?

Reply:
Good point, I'll shuffle that around.

Reply:
...and done.